code/game/objects/items/spear.dm

```diff
@@ -0,0 +1,169 @@
+				on_wield_callback=CALLBACK(src, .proc/on_wield), on_unwield_callback=CALLBACK(src, .proc/on_unwield))
+		throwforce = lancePart.throwforce
+		icon_prefix = lancePart.icon_prefix
+		parts_list -= G
+		parts_list -= lancePart
+		set_explosive(G)
+		qdel(lancePart)
+	..()
+
+/obj/item/spear/explosive/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to sword-swallow \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	user.say("[war_cry]", forced="spear warcry")
+	explosive.forceMove(user)
+	explosive.prime()
+	user.gib()
+	qdel(src)
+	return BRUTELOSS
+
+/obj/item/spear/explosive/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Alt-click to set your war cry.</span>"
+
+/obj/item/spear/explosive/AltClick(mob/user)
+	if(user.canUseTopic(src, BE_CLOSE))
+		..()
+		if(istype(user) && loc == user)
+			var/input = stripped_input(user,"What do you want your war cry to be? You will shout it when you hit someone in melee.", ,"", 50)
+			if(input)
+				src.war_cry = input
+
+/obj/item/spear/explosive/afterattack(atom/movable/AM, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+	if(wielded)
+		user.say("[war_cry]", forced="spear warcry")
+		explosive.forceMove(AM)
+		explosive.prime()
+		qdel(src)
+
+//GREY TIDE
+/obj/item/spear/grey_tide
+	name = "\improper Grey Tide"
+	desc = "Recovered from the aftermath of a revolt aboard Defense Outpost Theta Aegis, in which a seemingly endless tide of Assistants caused heavy casualities among Nanotrasen military forces."
+	attack_verb = list("gored")
+
+/obj/item/spear/grey_tide/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded=15, force_wielded=25, icon_update_callback=CALLBACK(src, .proc/icon_update_callback))
+
+/obj/item/spear/grey_tide/afterattack(atom/movable/AM, mob/living/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+	user.faction |= "greytide([REF(user)])"
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(istype (L, /mob/living/simple_animal/hostile/illusion))
+			return
+		if(!L.stat && prob(50))
+			var/mob/living/simple_animal/hostile/illusion/M = new(user.loc)
+			M.faction = user.faction.Copy()
+			M.Copy_Parent(user, 100, user.health/2.5, 12, 30)
+			M.GiveTarget(L)
+
+/*
+ * Bone Spear
+ */
+/obj/item/spear/bonespear	//Blatant imitation of spear, but made out of bone. Not valid for explosive modification.
+	icon_state = "bone_spear0"
+	name = "bone spear"
+	desc = "A haphazardly-constructed yet still deadly weapon. The pinnacle of modern technology."
+	force = 12
+	throwforce = 22
+	armour_penetration = 15				//Enhanced armor piercing
+
+/obj/item/spear/bonespear/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded=12, force_wielded=20, , icon_update_callback=CALLBACK(src, .proc/icon_update_callback))
+
+/obj/item/spear/bonespear/icon_update_callback(wielded)
```

Comment:
ditto

Reply:
bonespear is using the icon_update_callback and does not track wielded on item.

Reply:
It doesn't need to use a callback this specific case is better handled by a signal. Please stop resolving comments without actually adressing the feedback.

Reply:
now managed by icon_prefix

Reply:
this is managed in the component now with icon_prefix